libtransmission/torrent.cc

```diff
@@ -1191,7 +1191,7 @@ void torrentInit(tr_torrent* tor, tr_ctor const* ctor)
     {
         on_metainfo_completed(tor);
     }
-    else if (tor->start_when_stable || !has_metainfo)
```

Comment:
Agreed. That should be the correct flow.

Reply:
OK so you're saying revert b278c776117f5fc41ae18f415cbc2e9d573c856c, correct?

Reply:
I've just described the expectation.
I haven't tested the revert yet, but I believe that line is impacting it. I'll try the revert now.

Reply:
OK, I've tested the revert: it's back to a good behavior regarding the expectation of downloading metadata.
But at the same time, we're back to the issue described in my first comment in this pull request: closing-reopening the app with magnets with unfetched metadata will systematically unpause them.

Reply:
That doesn't seem to be consistent with what @sweetppro described at https://github.com/transmission/transmission/pull/4856#issuecomment-1427211841 and so I'm confused on what's actually going on -- He said that magnets with unfetched data will unpause on restart long enough to get the metadata, and then pause.

Does closing-reopening a paused magnet on restart unpause them for both metadata and download, or does it re-pause them after metadata?

And for the question above, @Coeur and @sweetppro are you two seeing the same behavior?

Reply:
I put a breakpoint, and I believe that this undesired behavior on macOS is caused by:
https://github.com/transmission/transmission/blob/25b50a7dffc1225465fd3cd45d0c22490b6886d4/macosx/Controller.mm#L746

That line may possibly force start all magnets on app launch...

Reply:
This is getting into a part of the macOS code that is probably better addressed by someone else. 

@Coeur @sweetppro if I revert https://github.com/transmission/transmission/commit/b278c776117f5fc41ae18f415cbc2e9d573c856c and merge can y'all take care of the remaining macOS issues?

Reply:
Ah no, sweetppro suggested change doesn't address it. The issue seems to be that `history["WaitToStart"]` is `false` after restarting the app (instead of being `true`).

Reply:
@ckerr is it possible to access start_when_stable from `tr_stat`?

Reply:
Actually, I'm getting closer to the culprit.
In `tr_torrentStartImpl`, when closing-reopening the app, the line `tr_isTorrent(tor)` is true even for magnets, so it goes to `tor->start_when_stable = true;`, and the magnet is unpaused.

So the issue might be that `tr_isTorrent(tor)` isn't behaving as expected for magnets when closing and reopening the app.

Reply:
No, it's not a second bug, I'm trying to analyze the regression caused by the present pull request.
This pull request is causing a regression on the macOS app. I'm simply trying to understand what's going on at 3 AM of the night (and I'll work at 7 AM), because you said you'd be merging this PR "tomorrow", even if it's causing regression.

I'm now looking at `tr_torrentStartImpl`. It forces `tor->start_when_stable` to be true. So when we close the app and reopen it, all magnets without metadata will be unpaused by this `tor->start_when_stable = true`. And it's not possible to read the value `start_when_stable` because it's in torrent.h and that file is forbidden to be imported by client.

Reply:
@Coeur what exactly is wrong with this fix:
```
if ([historyItem[@"Active"] boolValue] == NO)
         [torrent stopTransfer];
```

The active state applies to the paused state no?
It seems to in my testing.

On my machine (M2 Air), this will keep a magnet link (or any torrent for that matter) paused on restart, if it has been paused by the user. So the metadata will not be fetched. The PR provided already by @ckerr works for me in all other situations.

`WaitToStart` has no effect....

Im now not sure what is the expected behaviour, as you say metadata should always be downloaded, and the pause state should only apply to the actual torrent....

Reply:
Arg... you already merged it :( :(

Reply:
> 'm simply trying to understand what's going on at 3 AM of the night (and I'll go to work at 7 AM), because you said you'd be merging this PR "tomorrow", even if it's causing regression.

Don't make yoursef miserable on this! There's always tomorrow. We will hold off on 4.0.1 until all the dust settles!

Reply:
@sweetppro I put the explanation in my follow up pull request. The state "Active" and "WaitToStart" do not adequately reflect the magnet state because they are different from `start_when_stable`.